docs/best-practices.md

```diff
@@ -1,9 +1,52 @@
 # Type Registry
-**graphql-php** expects that each type in Schema is presented by single instance. Therefore
-if you define your types as separate PHP classes you need to ensure that each type is referenced only once.
- 
-Technically you can create several instances of your type (for example for tests), but `GraphQL\Type\Schema` 
-will throw on attempt to add different instances with the same name.
-
-There are several ways to achieve this depending on your preferences. We provide reference 
-implementation below that introduces TypeRegistry class:
+
+Every type in the schema must be represented by a single class instance.
+**graphql-php** throws when it discovers several instances with the same **name** in the schema.
+
+The typical way to do this is to create a registry of your types:
+
+```php
+<?php
+namespace MyApp;
+
+class TypeRegistry
+{
+    private $myAType;
+    private $myBType;
+    
+    public function myAType()
+    {
+        return $this->myAType ?: ($this->myAType = new MyAType($this));
+    }
+    
+    public function myBType()
+    {
+        return $this->myBType ?: ($this->myBType = new MyBType($this));
+    }
+}
+```
+
+Use this registry in type definitions:
+
+```php
+<?php
+namespace MyApp;
+use GraphQL\Type\Definition\ObjectType;
+
+class MyAType extends ObjectType
+{
+    public function __construct(TypeRegistry $types) 
+    {
+        parent::__construct([
+            'fields' => [
+                'b' => $types->myBType()                
+            ]
+        ]);
+    }
+}
+```
+Obviously, you can automate this registry as you wish to reduce boilerplate or even
```

Comment:
maybe add line break before

Reply:
That will be handled by `prettier` once we settled its setup in https://github.com/webonyx/graphql-php/pull/808.